service-mesh/config_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+)
+
+//go:embed golden/defaultEnvoy.json
+var defaultEnvoyConfig []byte
+
+func TestCompareEnvoyConfigToGolden(t *testing.T) {
+	require := require.New(t)
+
+	config, err := ParseProxyConfig("", "", "")
+	require.NoError(err)
+
+	testCases := map[string]struct {
+		pConfig ProxyConfig
+	}{
+		"success": {
+			pConfig: config,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+			configJSON, err := tc.pConfig.ToEnvoyConfig()
+			require.NoError(err)
+			fmt.Println(string(configJSON))
+			assert.JSONEq(string(defaultEnvoyConfig), string(configJSON))
+		})
+	}
+}
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
```

Comment:
Did you introduce this for a specific reason? This binary looks pretty serial...

Reply:
No, I thought we would just add this per convention for new tests, since e.g., https://github.com/edgelesssys/contrast/blob/5a6afdd360c22a30fbe336b1d56148623e33ec40/internal/fsstore/fsstore_test.go#L17 also looks pretty serial. It looks like at least Malte was of the same opinion, since all the times where this was also "unnecessarily" included are tests he wrote. But I don't have any (strong) opinion about this, does anyone else?

Reply:
I don't, was just curious.